tests/unit/models/command-test.js

```diff
@@ -558,6 +561,83 @@ describe('models/command.js', function() {
     });
   });
 
+  describe('runTask', function() {
+    let command;
+
+    class AsyncTask extends Task {
+      run(options) {
+        return new Promise(function(resolve) {
+          setTimeout(() => resolve(options), 1000);
+        });
+      }
+    }
+
+    class SyncTask extends Task {
+      run(options) { return options; }
+    }
+
+    class FailingTask extends Task {
+      run(options) { throw new Error('I was born to fail'); }
+    }
+
+    beforeEach(function() {
+      // this should be changed to new Command(), but needs more mocking
+      command = new ServeCommand(Object.assign({}, options, {
+        tasks: {
+          Async: AsyncTask,
+          Sync: SyncTask,
+          Failing: SyncTask,
+        },
+      }));
+    });
+
+    it('always handles task as a promise', function() {
+      command.runTask('Sync', { param: 'value' }).then(result => {
```

Comment:
we need to return promises in mocha tests no?

I think this should be `return command.runTask( ...`  to avoid flaky races in tests